pkg/custom_detectors/CUSTOM_DETECTORS.md

```diff
@@ -0,0 +1,209 @@
+# TruffleHog Custom Detector Setup Guide
+
+This guide will walk you through setting up a custom detector in TruffleHog to identify specific patterns unique to your project.
+
+## Steps to Set Up a Custom Detector
+
+1. **Create a Configuration File**:
+   - TruffleHog uses a configuration file, typically named `config.yaml`, to manage custom detector configuration.
+   - If this file doesn't exist, create it in your system.
+
+2. **Define the Custom Detector**:
+   - Open `config.yaml` with a text editor.
+   - Add a new detector under the `detectors` section.
+
+   Here's a template for a custom detector:
+
+   ```yaml
+   # config.yaml
+   detectors:
+     - name: HogTokenDetector
+       keywords:
+         - hog
+       regex:
+         token: '[^A-Za-z0-9+\/]{0,1}([A-Za-z0-9+\/]{40})[^A-Za-z0-9+\/]{0,1}'
```

Comment:
I chose to use a single regular expression to keep the example straightforward for beginners learning about custom detectors. Using two regexps makes the basic example complex to understand.